diff --git a/CMakeLists.txt b/CMakeLists.txt
index f3e69b8..575facd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -535,10 +535,20 @@ endif()
 
 add_subdirectory(unsupported)
 
-add_subdirectory(demos EXCLUDE_FROM_ALL)
+# Options to control configuration of optional components
+option(EIGEN_BUILD_DEMOS "Build demos (requires configuration)" OFF)
+option(EIGEN_BUILD_SCRIPTS "Build scripts (requires configuration)" OFF)
 
-# must be after test and unsupported, for configuring buildtests.in
-add_subdirectory(scripts EXCLUDE_FROM_ALL)
+if(EIGEN_BUILD_DEMOS)
+  add_subdirectory(demos EXCLUDE_FROM_ALL)
+endif()
+
+if(EIGEN_BUILD_SCRIPTS)
+  add_subdirectory(scripts EXCLUDE_FROM_ALL)
+endif()
+
+# Disable CTest to avoid dashboard targets
+# include(CTest)
 
 # TODO: consider also replacing EIGEN_BUILD_BTL by a custom target "make btl"?
 if(EIGEN_BUILD_BTL)
